cpp/ql/src/Security/CWE/CWE-014/MemsetMayBeDeleted.ql

```diff
@@ -0,0 +1,50 @@
+/**
+ * @name Call to `memset` may be deleted
+ * @description Using <code>memset</code> the function to clear private data in a variable that has no subsequent use
+ *              is potentially dangerous because the compiler can remove the call.
+ * @kind problem
+ * @id cpp/memset-may-be-deleted
+ * @problem.severity warning
```

Comment:
Since this is, and should be, a `warning` query, I'm not very happy with the CPP-Differences results. They're all of the form `char buf[N]; ...; memset(buf, 0, N);` with no other use of `buf` in the function. That's not clearing sensitive data, so I will go as far as saying these results are FPs. They could be TPs for a query that finds dead code, but that's not what we're doing here.

Reply:
I agree. I've relaxed the escaping restriction in 9e7c9d0ea091b8fdcf7fd43475b5cdb69477d67d so that we catch some actual true-positives, but I haven't made any attempt at removing the alerts on the `char buf[N]; ...; memset(buf, 0, N);` case. Should we exclude these cases as well?

Reply:
After an internal dialogue with myself over dinner, I came to the conclusion that we probably want to exclude these cases. So I did that in faadcd913e89a39226946678e30de844769e274b. [LGTM](https://lgtm.com/query/8103852947046036374/) only shows results in SHA-related functions now.